GLTFSDK.Samples/Serialize/Source/main.cpp

```diff
@@ -0,0 +1,294 @@
+        }
+
+        accessorIdPositions = bufferBuilder.AddAccessor(positions, { TYPE_VEC3, COMPONENT_FLOAT, false, std::move(minValues), std::move(maxValues) }).id;
+
+        // Add all of the Buffers, BufferViews and Accessors that were created using BufferBuilder to
+        // the Document. Note that after this point, no further calls should be made to BufferBuilder
+        bufferBuilder.Output(document);
+    }
+
+    void CreateTriangleEntities(Document& document, const std::string& accessorIdIndices, const std::string& accessorIdPositions)
+    {
+        // Create a very simple glTF Document with the following hierarchy:
+        //  Scene
+        //     Node
+        //       Mesh (Triangle)
+        //         MeshPrimitive
+        //           Material (Blue)
+        // 
+        // A Document can be constructed top-down or bottom up. However, if constructed top-down
+        // then the IDs of child entities must be known in advance, which prevents using the glTF
+        // SDK's automatic ID generation functionality.
+
+        // Construct a Material
+        Material material;
+        material.metallicRoughness.baseColorFactor = Color4(0.0f, 0.0f, 1.0f, 1.0f);
+        material.metallicRoughness.metallicFactor = 0.2f;
+        material.metallicRoughness.roughnessFactor = 0.4f;
+        material.doubleSided = true;
+
+        // Add it to the Document and store the generated ID
+        auto materialId = document.materials.Append(std::move(material), AppendIdPolicy::GenerateOnEmpty).id;
+
+        // Construct a MeshPrimitive. Unlike most types in glTF, MeshPrimitives are direct children
+        // of their parent Mesh entity rather than being children of the Document. This is why they
+        // don't have an ID member.
+        MeshPrimitive meshPrimitive;
+        meshPrimitive.materialId = materialId;
+        meshPrimitive.indicesAccessorId = accessorIdIndices;
+        meshPrimitive.attributes[ACCESSOR_POSITION] = accessorIdPositions;
+
+        // Construct a Mesh and add the MeshPrimitive as a child
+        Mesh mesh;
+        mesh.primitives.push_back(std::move(meshPrimitive));
+        // Add it to the Document and store the generated ID
+        auto meshId = document.meshes.Append(std::move(mesh), AppendIdPolicy::GenerateOnEmpty).id;
+
+        // Construct a Node adding a reference to the Mesh
+        Node node;
+        node.meshId = meshId;
+        // Add it to the Document and store the generated ID
+        auto nodeId = document.nodes.Append(std::move(node), AppendIdPolicy::GenerateOnEmpty).id;
+
+        // Construct a Scene
+        Scene scene;
+        scene.nodes.push_back(nodeId);
+        // Add it to the Document, using a utility method that also sets the Scene as the Document's default
+        document.SetDefaultScene(std::move(scene), AppendIdPolicy::GenerateOnEmpty);
+    }
+
+    void SerializeTriangle(std::experimental::filesystem::path path)
+    {
+        if (path.is_relative())
```

Comment:
Would prefer to move this path sanitization to the main method

Reply:
Done!